blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DAbstractMetricsRW.java

```diff
@@ -0,0 +1,259 @@
+        // in this loop, we will fire all the executeAsync() of
+        // various select statements, the collect all of the
+        // ResultSetFutures
+        Map<Locator, List<ResultSetFuture>> locatorToFuturesMap = new HashMap<Locator, List<ResultSetFuture>>();
+        Map<Locator, DAbstractMetricIO> locatorIOMap = new HashMap<Locator, DAbstractMetricIO>();
+
+        for (Locator locator : locators) {
+            try {
+
+                String rType = metadataCache.get( locator, MetricMetadata.ROLLUP_TYPE.name().toLowerCase() );
+
+                DAbstractMetricIO io = getIO( rType, granularity );
+
+                // put everything in a map of locator -> io so
+                // we can use em up later
+                locatorIOMap.put( locator, io );
+
+                // do the query
+                List<ResultSetFuture> selectFutures = io.selectForLocatorAndRange( columnFamily, locator, range );
+
+                // add all ResultSetFutures for a particular locator together
+                List<ResultSetFuture> existing = locatorToFuturesMap.get(locator);
+                if (existing == null) {
+                    existing = new ArrayList<ResultSetFuture>();
+                    locatorToFuturesMap.put(locator, existing);
+                }
+                existing.addAll(selectFutures);
+
+            } catch (CacheException ex) {
+                Instrumentation.markReadError();
+                LOG.error(String.format("Error looking up locator %s in cache", locator), ex);
+            }
+        }
+        return resultSetsToMetricData(locatorToFuturesMap, locatorIOMap, granularity);
+    }
+
+    /**
+     * Fetches a {@link com.rackspacecloud.blueflood.types.Points} object for a
+     * particular locator and rollupType from the specified column family and
+     * range
+     *
+     * @param locator
+     * @param rollupType
+     * @param range
+     * @param columnFamilyName
+     * @return
+     */
+    @Override
+    public Points getDataToRollup(final Locator locator,
+                                  RollupType rollupType,
+                                  Range range,
+                                  String columnFamilyName) throws IOException, CacheException {
+        Timer.Context ctx = Instrumentation.getReadTimerContext(columnFamilyName);
+        try {
+            // read the rollup object from the proper IO class
+            DAbstractMetricIO io = getIO( rollupType.name().toLowerCase(), CassandraModel.getGranularity( columnFamilyName ) );
+
+            Table<Locator, Long, Object> locatorTimestampRollup = io.getRollupsForLocator(locator, columnFamilyName, range);
+
+            // transform them to Points
+            Points points = new Points();
+            for (Table.Cell<Locator, Long, Object> cell : locatorTimestampRollup.cellSet()) {
+                points.add( createPoint( cell.getColumnKey(), cell.getValue()));
+            }
+            return points;
+        } finally {
```

Comment:
Looks like we don't do it on the astyanax-side, but it wouldn't hurt.  I'll add it.